FT5x06 driver build failure in C++ environment

### Board

Custom ESP32S3

### Hardware Description

Custom ESP32S3

### IDE Name

Espressif-IDE

### Operating System

Debian 13

### Description

When including the FT5x06 driver as a component from the ESP registry https://components.espressif.com/components/espressif/esp_lcd_touch_ft5x06 in a C++ project the ESP_LCD_TOUCH_IO_I2C_FT5x06_CONFIG() struct initialiser in https://github.com/espressif/esp-bsp/blob/master/components/lcd_touch/esp_lcd_touch_ft5x06/include/esp_lcd_touch_ft5x06.h has out of order fields which cause a compilation failure.

The error message:
`.../managed_components/espressif__esp_lcd_touch_ft5x06/include/esp_lcd_touch_ft5x06.h:56:5: error: designator order for field 'esp_lcd_panel_io_i2c_config_t::dev_addr' does not match declaration order in 'esp_lcd_panel_io_i2c_config_t'`

The struct in  should instead be
```#define ESP_LCD_TOUCH_IO_I2C_FT5x06_CONFIG()                \
    {                                                       \
        .dev_addr = ESP_LCD_TOUCH_IO_I2C_FT5x06_ADDRESS,    \
        .scl_speed_hz = 100000,                             \
        .control_phase_bytes = 1,                           \
        .dc_bit_offset = 0,                                 \
        .lcd_cmd_bits = 8,                                  \
        .flags =                                            \
        {                                                   \
            .disable_control_phase = 1,                     \
        }                                                   \
    }
```
Note that .dev_addr and .scl_speed_hz are in the opposite order in the BSP where the registry component is sourced from.
There are also missing fields in the .flags struct, but that doesn't cause a build failure.

### Sketch

```cpp
esp_lcd_panel_io_i2c_config_t tp_io_config = ESP_LCD_TOUCH_IO_I2C_FT5x06_CONFIG();
```

### Other Steps to Reproduce

_No response_

### I have checked existing issues, README.md and ESP32 Forum

- [x] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

--- a/.github/workflows/benchmark.yml
+++ b/.github/workflows/benchmark.yml
@@ -96,7 +96,7 @@ jobs:
           path: |
             ${{ env.BENCHMARK_PROJECT_DIR }}/build_*/benchmark_*.json
             ${{ env.BENCHMARK_PROJECT_DIR }}/build_*/size.json
-            *.log
+            /tmp/pytest-embedded/**/*.log
       - name: Upload benchmark metrics
         uses: actions/upload-artifact@v7
         if: always()

--- a/.github/workflows/build-run-applications.yml
+++ b/.github/workflows/build-run-applications.yml
@@ -169,4 +169,4 @@ jobs:
           name: ${{ env.TEST_RESULT_NAME }}
           path: |
             ${{ env.TEST_RESULT_FILE }}
-            *.log
+            /tmp/pytest-embedded/**/*.log

--- a/components/lcd_touch/esp_lcd_touch_cst816s/idf_component.yml
+++ b/components/lcd_touch/esp_lcd_touch_cst816s/idf_component.yml
@@ -1,4 +1,4 @@
-version: "1.1.1~1"
+version: "1.1.1~2"
 description: ESP LCD Touch CST816S - touch controller CST816S
 url: https://github.com/espressif/esp-bsp/tree/master/components/lcd_touch/esp_lcd_touch_cst816s
 dependencies:

--- a/components/lcd_touch/esp_lcd_touch_cst816s/include/esp_lcd_touch_cst816s.h
+++ b/components/lcd_touch/esp_lcd_touch_cst816s/include/esp_lcd_touch_cst816s.h
@@ -44,8 +44,8 @@ esp_err_t esp_lcd_touch_new_i2c_cst816s(const esp_lcd_panel_io_handle_t io, cons
  */
 #define ESP_LCD_TOUCH_IO_I2C_CST816S_CONFIG()             \
     {                                                     \
-        .scl_speed_hz = 100000,                           \
         .dev_addr = ESP_LCD_TOUCH_IO_I2C_CST816S_ADDRESS, \
+        .scl_speed_hz = 100000,                           \
         .on_color_trans_done = 0,                         \
         .user_ctx = 0,                                    \
         .control_phase_bytes = 1,                         \

--- a/components/lcd_touch/esp_lcd_touch_ft5x06/idf_component.yml
+++ b/components/lcd_touch/esp_lcd_touch_ft5x06/idf_component.yml
@@ -1,4 +1,4 @@
-version: "1.1.0~1"
+version: "1.1.0~2"
 description: ESP LCD Touch FT5x06 - touch controller FT5x06
 url: https://github.com/espressif/esp-bsp/tree/master/components/lcd_touch/esp_lcd_touch_ft5x06
 dependencies:

--- a/components/lcd_touch/esp_lcd_touch_ft5x06/include/esp_lcd_touch_ft5x06.h
+++ b/components/lcd_touch/esp_lcd_touch_ft5x06/include/esp_lcd_touch_ft5x06.h
@@ -44,8 +44,8 @@ esp_err_t esp_lcd_touch_new_i2c_ft5x06(const esp_lcd_panel_io_handle_t io, const
  */
 #define ESP_LCD_TOUCH_IO_I2C_FT5x06_CONFIG()                \
     {                                                       \
-        .scl_speed_hz = 100000,                             \
         .dev_addr = ESP_LCD_TOUCH_IO_I2C_FT5x06_ADDRESS,    \
+        .scl_speed_hz = 100000,                             \
         .control_phase_bytes = 1,                           \
         .dc_bit_offset = 0,                                 \
         .lcd_cmd_bits = 8,                                  \

--- a/components/lcd_touch/esp_lcd_touch_gt1151/idf_component.yml
+++ b/components/lcd_touch/esp_lcd_touch_gt1151/idf_component.yml
@@ -1,4 +1,4 @@
-version: "1.1.0~1"
+version: "1.1.0~2"
 description: ESP LCD Touch GT1151 - touch controller GT1151
 url: https://github.com/espressif/esp-bsp/tree/master/components/lcd_touch/esp_lcd_touch_gt1151
 dependencies:

--- a/components/lcd_touch/esp_lcd_touch_gt1151/include/esp_lcd_touch_gt1151.h
+++ b/components/lcd_touch/esp_lcd_touch_gt1151/include/esp_lcd_touch_gt1151.h
@@ -43,8 +43,8 @@ esp_err_t esp_lcd_touch_new_i2c_gt1151(const esp_lcd_panel_io_handle_t io, const
  */
 #define ESP_LCD_TOUCH_IO_I2C_GT1151_CONFIG()             \
     {                                                    \
-        .scl_speed_hz = 100000,                          \
         .dev_addr = ESP_LCD_TOUCH_IO_I2C_GT1151_ADDRESS, \
+        .scl_speed_hz = 100000,                          \
         .control_phase_bytes = 1,                        \
         .dc_bit_offset = 0,                              \
         .lcd_cmd_bits = 16,                              \

--- a/components/lcd_touch/esp_lcd_touch_gt911/idf_component.yml
+++ b/components/lcd_touch/esp_lcd_touch_gt911/idf_component.yml
@@ -1,4 +1,4 @@
-version: "1.2.0~2"
+version: "1.2.0~3"
 description: ESP LCD Touch GT911 - touch controller GT911
 url: https://github.com/espressif/esp-bsp/tree/master/components/lcd_touch/esp_lcd_touch_gt911
 dependencies:

--- a/components/lcd_touch/esp_lcd_touch_gt911/include/esp_lcd_touch_gt911.h
+++ b/components/lcd_touch/esp_lcd_touch_gt911/include/esp_lcd_touch_gt911.h
@@ -56,8 +56,8 @@ typedef struct {
  */
 #define ESP_LCD_TOUCH_IO_I2C_GT911_CONFIG()             \
     {                                                   \
-        .scl_speed_hz = 100000,                         \
         .dev_addr = ESP_LCD_TOUCH_IO_I2C_GT911_ADDRESS, \
+        .scl_speed_hz = 100000,                         \
         .control_phase_bytes = 1,                       \
         .dc_bit_offset = 0,                             \
         .lcd_cmd_bits = 16,                             \

--- a/components/lcd_touch/esp_lcd_touch_tt21100/idf_component.yml
+++ b/components/lcd_touch/esp_lcd_touch_tt21100/idf_component.yml
@@ -1,4 +1,4 @@
-version: "1.2.0~1"
+version: "1.2.0~2"
 description: ESP LCD Touch TT21100 - touch controller TT21100
 url: https://github.com/espressif/esp-bsp/tree/master/components/lcd_touch/esp_lcd_touch_tt21100
 dependencies:

--- a/components/lcd_touch/esp_lcd_touch_tt21100/include/esp_lcd_touch_tt21100.h
+++ b/components/lcd_touch/esp_lcd_touch_tt21100/include/esp_lcd_touch_tt21100.h
@@ -44,8 +44,8 @@ esp_err_t esp_lcd_touch_new_i2c_tt21100(const esp_lcd_panel_io_handle_t io, cons
  */
 #define ESP_LCD_TOUCH_IO_I2C_TT21100_CONFIG()               \
     {                                                       \
-        .scl_speed_hz = 100000,                             \
         .dev_addr = ESP_LCD_TOUCH_IO_I2C_TT21100_ADDRESS,   \
+        .scl_speed_hz = 100000,                             \
         .control_phase_bytes = 1,                           \
         .dc_bit_offset = 0,                                 \
         .lcd_cmd_bits = 16,                                 \

